lib/java/persistance/src/main/java/org/enso/persist/PerGenerator.java

```diff
@@ -59,6 +60,10 @@ final <T> int writeObject(T t) throws IOException {
     java.lang.Object obj = writeReplace.apply(t);
     java.lang.Integer found = knownObjects.get(obj);
     if (found == null) {
+      if (enteredObjects.put(obj, 0) != null) {
+        throw new IllegalStateException(
+            "Cyclic reference to " + obj.getClass().getCanonicalName() + " detected.");
+      }
```

Comment:
This is added as a placeholder until #9361 is implemented, to avoid nasty StackOverflow errors.

Reply:
OK.